Sources/NostrSDK/EventCreating.swift

```diff
@@ -466,4 +468,64 @@ public extension EventCreating {
 
         return try TimeBasedCalendarEvent(content: description, tags: tags, signedBy: keypair)
     }
+
+    /// Creates a ``CalendarNostrEvent`` (kind 31924), which is a collection of date-based and time-based calendar events.
+    /// - Parameters:
+    ///   - identifier: A unique identifier for the calendar. Can be reused in the future for replacing the calendar. If an identifier is not provided, a ``UUID`` string is used.
+    ///   - title: The title of the calendar.
+    ///   - description: A detailed description of the calendar.
+    ///   - calendarEventsCoordinates: The coordinates to date-based or time-based calendar events that belong to this calendar.
+    ///   - keypair: The Keypair to sign with.
+    /// - Returns: The signed ``CalendarNostrEvent``.
+    ///
+    /// See [NIP-52](https://github.com/nostr-protocol/nips/blob/master/52.md).
+    func calendarNostrEvent(withIdentifier identifier: String = UUID().uuidString, title: String, description: String = "", calendarEventsCoordinates: [EventCoordinates], signedBy keypair: Keypair) throws -> CalendarNostrEvent {
+        guard calendarEventsCoordinates.allSatisfy({ $0.kind == .dateBasedCalendarEvent || $0.kind == .timeBasedCalendarEvent }) else {
+            throw EventCreatingError.invalidInput
+        }
+
+        var tags: [Tag] = [
+            Tag(name: .identifier, value: identifier),
+            Tag(name: .title, value: title)
+        ]
+
+        calendarEventsCoordinates
+            .filter { $0.kind == .dateBasedCalendarEvent || $0.kind == .timeBasedCalendarEvent }
+            .forEach { tags.append($0.tag) }
+
+        return try CalendarNostrEvent(content: description, tags: tags, signedBy: keypair)
+    }
+
+    /// Creates a ``CalendarEventRSVP`` (kind 31925), which is a response to a calendar event to indicate a user's attendance intention.
+    /// - Parameters:
+    ///   - identifier: A unique identifier for the calendar event RSVP. Can be reused in the future for replacing the calendar event RSVP. If an identifier is not provided, a ``UUID`` string is used.
+    ///   - calendarEventCoordinates: The coordinates to date-based or time-based calendar event being responded to.
+    ///   - status: The attendance status to the referenced calendar event.
+    ///   - freebusy: Determines if the user would be free or busy for the duration of the calendar event. This tag must be omitted or ignored if the status label is set to declined.
```

Comment:
This "freebusy" stuff is confusing to me. What is the use-case of this?

Reply:
It mimics the iCal spec that traditional calendar clients like Google Calendar implement. It marks your availability. If you wanted to schedule time with someone, you would use that as the indicator if that person would be available.

Reply:
Nice. Maybe that link could be included in DocC comments somewhere.